### PR TITLE
Move command line --vacuum into xml

### PR DESCRIPTION
**Pull request by ye-luo**
As discussed in #146 #542 #578 
Deprecate --vacuum
In xml, use vac_scale. Since I often mistype 'vacuum', use only 'vac'
```
<simulationcell>
  <parameter name="bconds">
    p p n
  </parameter>
  <parameter name="vac_scale" > 1.2 </parameter>
</simulationcell>
```
Some additional change.
in PPN case,
The old code does `R(2,2)*=Scale`
The PR rescale the whole R(2) vector, namely
```
R(2,0)*=Scale;
R(2,1)*=Scale;
R(2,2)*=Scale;
```
in PNN case,
Both R(1) and R(2) are rescaled instead of only R(1,1) and R(2,2)

This should provide better support for the "N" directions not orthogonal to P direction.

I check the existence of --vacuum and the code is stopped to prevent wasting hours.

### CODE REVIEW

**Comment by ye-luo** [comment]
I don't know the historic reason but with the current choice, the rescaling is applied on R(2) not the z axis of R(2). So a pre-factor is more convenient than 3 distances for the vector.
in a wire case, vacuum is applied on R(1) and R(2) so put zaxis in the name is misleading.

**Comment by ye-luo** [comment]
1, abort has been handled properly.
2, input tag name changed to vacuum.
3, unit test added.
4, manual updated.
Ready to go.

**Comment by markdewing** [comment]
Should deprecation be a little more gradual? 
First step would be to add the new keyword handling and print a warning message if the command line option is used (but not stop)
Second step (in a future release) would be to print an error and stop of the command line option is used.

Maybe for this feature the cost of forcing the users to change their input files is not so high (versus the development cost of doing it more gradually)?

File: manual/simulationcell.tex [hunk 1]
```diff
@@ -26,17 +27,18 @@ \section{Specifying the simulation cell}
...
+  <simulationcell>
+    <parameter name="lattice">
+      3.80000000       0.00000000       0.00000000
+      0.00000000       3.80000000       0.00000000
+      0.00000000       0.00000000       3.80000000
+    </parameter>
+    <parameter name="bconds">
+       p p p
+    </parameter>
+    <!--parameter name="vacuum"> 1.0 </parameter-->
```
Comment (markdewing) [inline manual/simulationcell.tex]: I don't think this is clear, being put in here with an example that won't use the 'vacuum' option even if it is uncommented.   It should be put in a separate example with p p n or p n n boundary conditions.

**Comment by prckent** [comment]
@markdewing This is a good point about deprecation, but I think only the core developers have been using this feature. Sadly, it was not e.g. described in any tutorials.

File: manual/simulationcell.tex (hunk 1 above)
Comment (ye-luo) [reply manual/simulationcell.tex]: You are right. I will add an example in the vacuum section.

**Comment by ye-luo** [comment]
I feel no one reads (or reads carefully) the standard output file unless the code aborts.
If we do slow deprecation, users will not notice this until the actual deprecation happens.

**Comment by markdewing** [comment]
I agree that realistically no one will likely read the message and act on it until forced.   However, it might be nice to have a version of the that accepts the option both ways (?)

**Comment by ye-luo** [comment]
In principle, we should provide a version accepting both options and it is not difficult but messy.
In practice, I don't see enough needs and prefer just have this hard switch. We can test it cleanly and I don't need to revisit it.

**Pull request merged**